drivers/net/loopback.c

```diff
@@ -102,11 +111,27 @@ static int loopback_send(const struct device *dev, struct net_pkt *pkt)
 		res = -ENOMEM;
 		goto out;
 	}
-
-	res = net_recv_data(net_pkt_iface(cloned), cloned);
-	if (res < 0) {
-		LOG_ERR("Data receive failed.");
+	
+#ifdef CONFIG_NET_LOOPBACK_PACKET_DROP
+	/* Drop packets based on the loopback_packet_drop_ratio
+	a ratio of 0.2 will drop one every 5 packets */
+	loopback_packet_drop_state += loopback_packet_drop_ratio;
+	if (loopback_packet_drop_state < 1.0f) {
```

Comment:
If you reverse the logic here (`if (loopback_packet_drop_state  >= 1.0f) {`) you could have a single, cocise `#ifdef` block. It could end up with `goto out;` so you won't even need to modify existing code.

Reply:
Thanks for the suggestion, done